src/wp-includes/rest-api/endpoints/class-wp-rest-abilities-v1-categories-controller.php

```diff
@@ -0,0 +1,293 @@
+<?php
+/**
+ * REST API ability categories controller for Abilities API.
+ *
+ * @package WordPress
+ * @subpackage Abilities_API
+ * @since 6.9.0
+ */
+
+declare( strict_types = 1 );
+
+/**
+ * Core controller used to access ability categories via the REST API.
+ *
+ * @since 6.9.0
+ *
+ * @see WP_REST_Controller
+ */
+class WP_REST_Abilities_V1_Categories_Controller extends WP_REST_Controller {
+
+	/**
+	 * REST API namespace.
+	 *
+	 * @since 6.9.0
+	 * @var string
+	 */
+	protected $namespace = 'wp-abilities/v1';
+
+	/**
+	 * REST API base route.
+	 *
+	 * @since 6.9.0
+	 * @var string
+	 */
+	protected $rest_base = 'categories';
+
+	/**
+	 * Registers the routes for ability categories.
+	 *
+	 * @since 6.9.0
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes(): void {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<slug>[a-z0-9]+(?:-[a-z0-9]+)*)',
+			array(
+				'args'   => array(
+					'slug' => array(
+						'description' => __( 'Unique identifier for the ability category.' ),
+						'type'        => 'string',
+						'pattern'     => '^[a-z0-9]+(?:-[a-z0-9]+)*$',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Retrieves all ability categories.
+	 *
+	 * @since 6.9.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response Response object on success.
+	 */
+	public function get_items( $request ) {
+		$categories = wp_get_ability_categories();
+
+		$page     = $request['page'];
+		$per_page = $request['per_page'];
+		$offset   = ( $page - 1 ) * $per_page;
+
+		$total_categories = count( $categories );
+		$max_pages        = ceil( $total_categories / $per_page );
```

Comment:
Better to cast int here

Reply:
Ok, I see other occurrences use `(int)`. Good catch, thank you 👍🏻